server/src/controllers/redeemCode.js

```diff
@@ -0,0 +1,46 @@
+const { HttpError } = require('../utils');
+const { DiscountCode, UserDiscounts } = require('../models');
+
+// eslint-disable-next-line consistent-return
+exports.redeemCode = async (req, res, next) => {
+  const { promocode, totalPrice } = req.body;
+  let newPrice;
+  try {
+    const code = await DiscountCode.findOne({
+      where: { promocode },
+    });
+
+    if (!code)
+      throw new HttpError(
+        400,
+        'The code you entered is incorrect, please try again.'
+      );
+
+    const { uid } = req.user;
+    const promocodeId = code.id;
+
+    const data = await UserDiscounts.findOrCreate({
+      where: { promocode: promocodeId, userid: uid },
+    });
+
+    if (!data[1])
```

Comment:
data[1] or data[0]?

Reply:
data[0] contains the information, the second index indicates whether the new id has been added by ( findOrCreate ) method or already exits.